source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/typeaheadsearch/TypeAheadSearchViewModel.kt

```diff
@@ -0,0 +1,200 @@
+                        _uiState.postValue(DynamicTypeAheadUiState.ShowingChoices)
+                    else
+                        _uiState.postValue(DynamicTypeAheadUiState.NoResults)
+
+
+                    adapter.setChoices(titles, values)
+                }
+                else if (_queryText.value.equals(queryText)) {
+                    adapter.setChoices(ArrayList(), ArrayList())
+                    _uiState.postValue(DynamicTypeAheadUiState.Error)
+                }
+            }
+        }
+    }
+
+    class FilteredAdapter (private var m_choices: List<String>, private var m_values: List<String>) : RecyclerView.Adapter<FilteredAdapter.ViewHolder>() {
+
+        @SuppressLint("NotifyDataSetChanged")
+        fun setChoices(choices: List<String>, values: List<String>) {
+            m_choices = choices
+            m_values = values
+            notifyDataSetChanged()
+        }
+        // Provide a direct reference to each of the views within a data item
+        // Used to cache the views within the item layout for fast access
+        inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+            // Your holder should contain and initialize a member variable
+            // for any view that will be set as you render a row
+            val textView: TextView = itemView.findViewById(android.R.id.text1)
+        }
+
+        // ... constructor and member variables
+        // Usually involves inflating a layout from XML and returning the holder
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FilteredAdapter.ViewHolder {
+            val context = parent.context
+            val inflater = LayoutInflater.from(context)
+            // Inflate the custom layout
+            val choicesView = inflater.inflate(R.layout.adaptive_card_select_item, parent, false)
+            // Return a new holder instance
+            return ViewHolder(choicesView)
+        }
+
+        // Involves populating data into the item through holder
+        override fun onBindViewHolder(viewHolder: FilteredAdapter.ViewHolder, position: Int) {
+            // Get the data model based on position
+            val title: String = m_choices[position]
+            val value: String = m_values[position]
+            // Set item views based on your views and data model
+            val textView = viewHolder.textView
+            textView.text = title
+            textView.setOnClickListener {
+                if (it.context is Activity) {
+                    it.postDelayed({
+                        val activity: Activity = it.context as Activity
+                        val intent = activity.intent
+                        intent.putExtra("typeAheadSearchSelectedKey", title)
+                        intent.putExtra("typeAheadSearchSelectedValue", value)
+                        activity.setResult(Activity.RESULT_OK, intent)
+                        activity.finish()
+                    }, 100)
```

Comment:
>100 [](http://example.com/codeflow?start=23&length=3)

can you add a comment next to this number to understand what is it referring to? I would assume it's a delay time or a timeout value but it would ease the readability if we have to fix something

Reply:
added